### PR TITLE
Dialog boxes are now modal

### PR DESCRIPTION
**Pull request by peterhinch**
Formerly if the user clicked on the main window when a dialog box was present, the dialog became hidden behind the main window. To a naive user it appeared as if the application had crashed. This patch ensures dialogs remain visible.

### CODE REVIEW

**Comment by ZanderBrown** [comment]
this is for issue #128

File: mu/contrib/microfs.py [hunk 1]
```diff
@@ -41,13 +41,13 @@
 """
 
 
-def find_microbit():
+def find_upython_device():
```
Comment (carlosperate) [inline mu/contrib/microfs.py]: Looks like this might be a left over from the esp8266 fork.

**Comment by carlosperate** [comment]
I was thinking, `self._view` is in instance of the `Window` class and the `QMessageBox`
is created within the method `Window.show_message()`. Instead of having to indicate a parent widget in the logic portion of the application, could we instead just pass `self` as the parent for `QMessageBox`? As we want the main window to always be the parent for anything opened by `Window.show_message()`.

File: mu/contrib/microfs.py (hunk 1 above)
Comment (peterhinch) [inline mu/contrib/microfs.py]: You're right. I'm afraid I've got hopelessly confused with Git/Github.

**Comment by ZanderBrown** [comment]
i don't see why not

**Comment by ntoll** [comment]
Hi!

Thanks for the contribution. I notice Travis has had a fit of the collywobbles. Care to revise? This is a valuable contribution!

:-)

N.

**Comment by ZanderBrown** [comment]
As mentioned in #128 I was going to be putting together a cleaner PR on @peterhinch 's behalf although I might modify taking into account @carlosperate 's comment. I will admit I forgot but it's top of my TODO for tomorrow morning.

**Comment by ntoll** [comment]
@ZanderBrown cheers. Make sure you reference this PR as the thing being replaced. ;-)

**Comment by carlosperate** [comment]
As this fix has been already merged in #141 we can close this PR.

**Pull request closed without merging**